include/drivers/clock_control.h

```diff
@@ -150,6 +151,39 @@ static inline int clock_control_off(const struct device *dev,
 	return api->off(dev, sys);
 }
 
+/**
+ * @brief Configure a source clock
+ *
+ * This function is non-blocking and can be called from any context.
+ * On success, the selected clock is configured as the client's source clock.
+ *
+ * It is client's reposnsibility to ensure that subsequent calls to the API
+ * provides the right information to enable clock_control driver to perform
+ * the right action (such as using the right clock source on _get_rate call).
+ *
+ * @param dev Device structure whose driver controls the clock
+ * @param sys Opaque data representing the clock
+ * @return 0 on success, negative errno on failure.
+ */
+static inline int clock_control_select(const struct device *dev,
```

Comment:
since brief states "Configure a source clock" maybe better would be `clock_control_configure`?

I would also add `void *data` parameter. It will allow vendor specific features, e.g. frequency reconfiguration.

Reply:
I used `select` on purpose as the proposed scope was limited to source clock selection.
Expanding the scope makes sense, but then should we define a set of actions for which drivers should discriminate support ?

Reply:
imo, we don't need to define any actions. Truth is that clock control api is not vendor agnostic, `subsys` argument is vendor specific. I didn't see any code where clock_control API would be used in generic code. I think that API is mainly to provide uniformed look & feel and access to clocks and not to provide abstraction.

Reply:
Valid. Let me update the proposal.